### PR TITLE
NEXTPY-2214_Enabled-complete-validation-skip

### PR DESCRIPTION
**Pull request by paulwispnext**
Added skip validations to immediately skip the whole validation process. 
(where the options just allow you to skip specific parts of specific validations)

### CODE REVIEW

**Review by ISPnextHarm — APPROVED** [review]
Code review OK 
Checked for potential security issues OK
Checked for code optimization OK
Checked for code duplication OK
Checked for whether names are descriptive enough OK
Checked for possible performance improvements OK
Checked for the presence and quality of tests OK
Checked for additional code documentation OK
Checked for backwards compatibility OK

File: src/state.ts [hunk 1]
```diff
@@ -418,21 +418,26 @@ export class FormState<
...
+        extraOptions = { ignoreGetError: true };
+      }
+      const isValid = this.validate({ ...extraOptions, ...options });
 
-    if (!options.ignoreSaveStatus) {
-      this.setSaveStatus("rightAfter");
-    }
-    if (!isValid) {
-      return false;
+      if (!options.ignoreSaveStatus) {
```
Comment (randywanga) [inline src/state.ts:435]: Should this be inside the skipValidations if-block or outside of it?

Comment (paulwispnext) [reply src/state.ts:435]: I think yes, as the "rightAfter" means "right after validation" - we do not use validation so we shouldn't go to that state right?

**Review by kdleijer — APPROVED** [review]
Code review: 
- Checked for potential security issues 
- Checked for code optimisation 
- Checked for code duplication 
- Checked for whether names are descriptive enough 
- Checked for possible performance improvements 
- Checked for the presence and quality of tests 
- Checked for additional code documentation 
- Checked for backwards compatibility               

Seems harmless since you've made sure that;
- skipValidations is optional and defaulted to false
- all original behaviour is the same as long as skipValidations===False

**Pull request merged**